### PR TITLE
[FEATURE] implemented new widget for sidebar

### PR DESCRIPTION
**Pull request by michaelzangerle**
**Tasks:**
- [x] gather feedback for my changes

**Informations:**

| Q | A |
| --- | --- |
| Tests pass? | - |
| Fixed tickets | - |
| BC Breaks | - |
| Doc | - |

**Requires:**
- https://github.com/sulu-cmf/SuluSalesCoreBundle/pull/7
- https://github.com/sulu-cmf/SuluSalesShippingBundle/pull/4

**Issues:**
- https://github.com/massiveart/POOL-ALPIN/issues/182

### CODE REVIEW

File: Api/Order.php [hunk 1]
```diff
@@ -731,4 +732,40 @@ public function getOrderDate()
...
+    }
+
+    /**
+     * Returns object as array
+     *
+     * @return array []
+     */
+    public function toArray()
+    {
+
```
Comment (turbo-ele) [inline Api/Order.php]: remove empty line

File: Resources/public/js/components/orders/components/form/main.js [hunk 2]
```diff
@@ -575,11 +575,24 @@ define([
...
 
             if(!!data.contact && !!data.account && !!data.status){
-                url += '?contact='+data.contact.id+'&account='+data.account.id+'&status='+data.status.status;
+                uriTemplate = this.sandbox.uritemplate.parse(link);
+                url = uriTemplate.expand({
+                    params: {
+                        contact: data.contact.id,
+                        account: data.account.id,
+                        status: data.status.status,
+                        locale: SULU.user.locale,
```
Comment (turbo-ele) [inline Resources/public/js/components/orders/components/form/main.js]: isn't there a function for getting global vars?

File: Widgets/FlowOfDocuments.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+     *
+     * @param array $options
+     * @throws WidgetException
+     * @return array
+     */
+    public function getData($options)
+    {
+        if ($this->checkRequiredParameters($options)) {
+            $this->getOrderData($options);
+            $this->getShippingData($options);
```
Comment (turbo-ele) [inline Widgets/FlowOfDocuments.php]: the function names getOrderData and getShippingData are a little bit misleading, since they add entries to the class

Comment (michaelzangerle) [inline Widgets/FlowOfDocuments.php]: @turbo-ele getData() is inherited from the WidgetInterface - will rename getShippingData to fetch ...

**Pull request merged**